File: src/main/java/com/polykhel/billnext/service/CategoryService.java
package com.polykhel.billnext.service;

import com.polykhel.billnext.service.dto.CategoryDTO;
import java.util.Optional;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

/**
 * Service Interface for managing {@link com.polykhel.billnext.domain.Category}.
 */
public interface CategoryService {
    /**
     * Save a category.
     *
     * @param categoryDTO the entity to save.
     * @return the persisted entity.
     */
    CategoryDTO save(CategoryDTO categoryDTO);

    /**
     * Partially updates a category.
     *
     * @param categoryDTO the entity to update partially.
     * @return the persisted entity.
     */
    Optional<CategoryDTO> partialUpdate(CategoryDTO categoryDTO);

    /**
     * Get all the categories.
     *
     * @param pageable the pagination information.
     * @return the list of entities.
     */
    Page<CategoryDTO> findAll(Pageable pageable);

    /**
     * Get the "id" category.
     *
     * @param id the id of the entity.
     * @return the entity.
     */
    Optional<CategoryDTO> findOne(Long id);

    /**
     * Delete the "id" category.
     *
     * @param id the id of the entity.
     */
    void delete(Long id);

    /**
     * Get all the categories by the current user.
     *
     * @param pageable the pagination information.
     * @return the list of entities.
     */
    Page<CategoryDTO> findAllByCurrentUser(Pageable pageable);
}

File: src/main/kotlin/com/polykhel/billnext/service/CategoryService.kt
package com.polykhel.billnext.service

import com.polykhel.billnext.service.dto.CategoryDTO
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import java.util.*

/**
 * Service Interface for managing [com.polykhel.billnext.domain.Category].
 */
interface CategoryService {

    /**
     * Save a category.
     *
     * @param categoryDTO the entity to save.
     * @return the persisted entity.
     */
    fun save(categoryDTO: CategoryDTO): CategoryDTO

    /**
     * Partially updates a category.
     *
     * @param categoryDTO the entity to update partially.
     * @return the persisted entity.
     */
    fun partialUpdate(categoryDTO: CategoryDTO): Optional<CategoryDTO>

    /**
     * Get all the categories.
     *
     * @param pageable the pagination information.
     * @return the list of entities.
     */
    fun findAll(pageable: Pageable): Page<CategoryDTO>

    /**
     * Get all the categories by currentUser.
     *
     * @param pageable the pagination information.
     * @return the list of entities.
     */
    fun findAllByCurrentUser(pageable: Pageable): Page<CategoryDTO>

    /**
     * Get the "id" category.
     *
     * @param id the id of the entity.
     * @return the entity.
     */
    fun findOne(id: Long): Optional<CategoryDTO>

    /**
     * Delete the "id" category.
     *
     * @param id the id of the entity.
     */
    fun delete(id: Long)
}
